### ISSUE TITLE
Issue on IOS device since 23Feb 2021

### ISSUE THREAD

**Issue by UjjwaRaijada** [ShekarMudaliyar/social_share#47]
Xcode's output:
↳
    1 warning generated.
    /Users/apple/Developer/flutter/.pub-cache/hosted/pub.dartlang.org/social_share-2.0.5/ios/Classes/SocialSharePlugin.m:136:45: warning: 'stringByAddingPercentEscapesUsingEncoding:' is deprecated: first deprecated in iOS 9.0 - Use -stringByAddingPercentEncodingWithAllowedCharacters: instead, which always uses the recommended UTF-8 encoding, and which encodes for a specific URL component or subcomponent since each URL component or subcomponent has different rules for what characters are valid. [-Wdeprecated-declarations]
          NSString* urlTextEscaped = [urlstring stringByAddingPercentEscapesUsingEncoding:NSUTF8StringEncoding];
                                                ^
    In module 'UIKit' imported from /Users/apple/AndroidStudioProjects/flutter_app_like4like/ios/Pods/Target Support Files/social_share/social_share-prefix.pch:2:
    In module 'Foundation' imported from /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator14.4.sdk/System/Library/Frameworks/UIKit.framework/Headers/UIKit.h:8:
    /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator14.4.sdk/System/Library/Frameworks/Foundation.framework/Headers/NSURL.h:595:1: note: 'stringByAddingPercentEscapesUsingEncoding:' has been explicitly marked deprecated here
    - (nullable NSString *)stringByAddingPercentEscapesUsingEncoding:(NSStringEncoding)enc API_DEPRECATED("Use -stringByAddingPercentEncodingWithAllowedCharacters: instead, which always uses the recommended UTF-8 encoding, and which encodes for a specific URL component or subcomponent since each URL component or subcomponent has different rules for what characters are valid.", macos(10.0,10.11), ios(2.0,9.0), watchos(2.0,2.0), tvos(9.0,9.0));
    ^
    /Users/apple/Developer/flutter/.pub-cache/hosted/pub.dartlang.org/social_share-2.0.5/ios/Classes/SocialSharePlugin.m:195:47: warning: 'stringByAddingPercentEscapesUsingEncoding:' is deprecated: first deprecated in iOS 9.0 - Use -stringByAddingPercentEncodingWithAllowedCharacters: instead, which always uses the recommended UTF-8 encoding, and which encodes for a specific URL component or subcomponent since each URL component or subcomponent has different rules for what characters are valid. [-Wdeprecated-declarations]
            NSString* urlTextEscaped = [urlstring stringByAddingPercentEscapesUsingEncoding:NSUTF8StringEncoding];
                                                  ^
    In module 'UIKit' imported from /Users/apple/AndroidStudioProjects/flutter_app_like4like/ios/Pods/Target Support Files/social_share/social_share-prefix.pch:2:
    In module 'Foundation' imported from /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator14.4.sdk/System/Library/Frameworks/UIKit.framework/Headers/UIKit.h:8:
    /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator14.4.sdk/System/Library/Frameworks/Foundation.framework/Headers/NSURL.h:595:1: note: 'stringByAddingPercentEscapesUsingEncoding:' has been explicitly marked deprecated here
    - (nullable NSString *)stringByAddingPercentEscapesUsingEncoding:(NSStringEncoding)enc API_DEPRECATED("Use -stringByAddingPercentEncodingWithAllowedCharacters: instead, which always uses the recommended UTF-8 encoding, and which encodes for a specific URL component or subcomponent since each URL component or subcomponent has different rules for what characters are valid.", macos(10.0,10.11), ios(2.0,9.0), watchos(2.0,2.0), tvos(9.0,9.0));
    ^
    /Users/apple/Developer/flutter/.pub-cache/hosted/pub.dartlang.org/social_share-2.0.5/ios/Classes/SocialSharePlugin.m:265:67: warning: 'stringByAddingPercentEscapesUsingEncoding:' is deprecated: first deprecated in iOS 9.0 - Use -stringByAddingPercentEncodingWithAllowedCharacters: instead, which always uses the recommended UTF-8 encoding, and which encodes for a specific URL component or subcomponent since each URL component or subcomponent has different rules for what characters are valid. [-Wdeprecated-declarations]
                 NSURL * whatsappURL = [NSURL URLWithString:[urlWhats stringByAddingPercentEscapesUsingEncoding:NSUTF8StringEncoding]];
                                                                      ^
    In module 'UIKit' imported from /Users/apple/AndroidStudioProjects/flutter_app_like4like/ios/Pods/Target Support Files/social_share/social_share-prefix.pch:2:
    In module 'Foundation' imported from /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator14.4.sdk/System/Library/Frameworks/UIKit.framework/Headers/UIKit.h:8:
    /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator14.4.sdk/System/Library/Frameworks/Foundation.framework/Headers/NSURL.h:595:1: note: 'stringByAddingPercentEscapesUsingEncoding:' has been explicitly marked deprecated here
    - (nullable NSString *)stringByAddingPercentEscapesUsingEncoding:(NSStringEncoding)enc API_DEPRECATED("Use -stringByAddingPercentEncodingWithAllowedCharacters: instead, which always uses the recommended UTF-8 encoding, and which encodes for a specific URL component or subcomponent since each URL component or subcomponent has different rules for what characters are valid.", macos(10.0,10.11), ios(2.0,9.0), watchos(2.0,2.0), tvos(9.0,9.0));
    ^
    /Users/apple/Developer/flutter/.pub-cache/hosted/pub.dartlang.org/social_share-2.0.5/ios/Classes/SocialSharePlugin.m:267:87: error: expected expression
                     [[UIApplication sharedApplication] openURL:options:completionHandler:: whatsappURL];
                                                                                          ^
    /Users/apple/Developer/flutter/.pub-cache/hosted/pub.dartlang.org/social_share-2.0.5/ios/Classes/SocialSharePlugin.m:277:76: warning: 'stringByAddingPercentEscapesUsingEncoding:' is deprecated: first deprecated in iOS 9.0 - Use -stringByAddingPercentEncodingWithAllowedCharacters: instead, which always uses the recommended UTF-8 encoding, and which encodes for a specific URL component or subcomponent since each URL component or subcomponent has different rules for what characters are valid. [-Wdeprecated-declarations]
                         NSURL * telegramURL = [NSURL URLWithString:[urlScheme stringByAddingPercentEscapesUsingEncoding:NSUTF8StringEncoding]];
                                                                               ^
    In module 'UIKit' imported from /Users/apple/AndroidStudioProjects/flutter_app_like4like/ios/Pods/Target Support Files/social_share/social_share-prefix.pch:2:
    In module 'Foundation' imported from /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator14.4.sdk/System/Library/Frameworks/UIKit.framework/Headers/UIKit.h:8:
    /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator14.4.sdk/System/Library/Frameworks/Foundation.framework/Headers/NSURL.h:595:1: note: 'stringByAddingPercentEscapesUsingEncoding:' has been explicitly marked deprecated here
    - (nullable NSString *)stringByAddingPercentEscapesUsingEncoding:(NSStringEncoding)enc API_DEPRECATED("Use -stringByAddingPercentEncodingWithAllowedCharacters: instead, which always uses the recommended UTF-8 encoding, and which encodes for a specific URL component or subcomponent since each URL component or subcomponent has different rules for what characters are valid.", macos(10.0,10.11), ios(2.0,9.0), watchos(2.0,2.0), tvos(9.0,9.0));
    ^
    /Users/apple/Developer/flutter/.pub-cache/hosted/pub.dartlang.org/social_share-2.0.5/ios/Classes/SocialSharePlugin.m:267:61: error: use of undeclared identifier 'options'
                     [[UIApplication sharedApplication] openURL:options:completionHandler:: whatsappURL];
                                                                ^
    /Users/apple/Developer/flutter/.pub-cache/hosted/pub.dartlang.org/social_share-2.0.5/ios/Classes/SocialSharePlugin.m:267:69: error: use of undeclared identifier 'completionHandler'
                     [[UIApplication sharedApplication] openURL:options:completionHandler:: whatsappURL];
                                                                        ^
    4 warnings and 3 errors generated.
    note: Using new build system
    note: Building targets in parallel
    note: Planning build
    note: Constructing build description

**Comment by Shvet** [comment]
did you get any solution, i am getting same error

**Comment by UjjwaRaijada** [comment]
> did you get any solution, i am getting same error

No. I did not. I changed the package. I am using Share "https://pub.dev/packages/share". It is working perfect.

**Comment by Samir-Shaikh** [comment]
Anyone get a solution? I am really tried everything please help if anyone have solution.
